pkg/controller/jobframework/reconciler.go

```diff
@@ -780,12 +834,22 @@ func (r *JobReconciler) getPodSetsInfoFromStatus(ctx context.Context, w *kueue.W
 func (r *JobReconciler) handleJobWithNoWorkload(ctx context.Context, job GenericJob, object client.Object) error {
 	log := ctrl.LoggerFrom(ctx)
 
+	// Stop the job if not already suspended
+	if stopErr := r.stopJob(ctx, job, nil, StopReasonNoMatchingWorkload, "missing workload"); stopErr != nil {
```

Comment:
I'm concerned with this change as it calls stopJob unconditionally, so affects also the `!usePrebuiltWorkload` scenario. Can we check if `!usePrebuiltWorkload` prior to calling it?

Reply:
Before the need for prebuiltWorkload, the creation of the new workloads wold have trigger a new reconcile just after the current one in which the job wold have been stopped. We are doing the same thing , just one reconcile call earlier.

Reply:
Hm, before, if `job.IsActive()` returned `false`, then we created the workload lower in this function, without calling the `stopJob` at all. 

It is not very easy to assess potential nuances of the change on the old code paths. I think surrounding this call with `usePrebuiltWorkload` would make it safer, so that we don't modify the regular code paths.

Reply:
done